third_party/CMakeLists.txt

```diff
@@ -150,4 +150,20 @@ target_link_libraries(pqxx PUBLIC ${PQ_LIBRARY})
 
 add_library(uninitialized_vector INTERFACE)
 target_include_directories(uninitialized_vector INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/uninitialized_vector/)
-target_sources(uninitialized_vector INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/uninitialized_vector/uninitialized_vector.hpp)
\ No newline at end of file
+target_sources(uninitialized_vector INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/uninitialized_vector/uninitialized_vector.hpp)
+
+include(ExternalProject)
+externalproject_add(
+    libjemalloc
+
+    PREFIX ${CMAKE_CURRENT_SOURCE_DIR}/jemalloc
+    URL https://github.com/jemalloc/jemalloc/releases/download/5.1.0/jemalloc-5.1.0.tar.bz2
+    # Executing the configure script with suppressed output, only printing it in case of non-zero return
+    CONFIGURE_COMMAND bash -c "cd <SOURCE_DIR> && output=`./configure 2>&1` || printf \"$output\\n\""
+    BUILD_COMMAND ${CMAKE_BUILD_TOOL}
+    BUILD_IN_SOURCE 1
+    STEP_TARGETS build
+)
+add_library(jemalloc SHARED IMPORTED GLOBAL)
+add_dependencies(jemalloc libjemalloc-build)
+set_target_properties(jemalloc PROPERTIES IMPORTED_LOCATION ${CMAKE_CURRENT_SOURCE_DIR}/jemalloc/src/libjemalloc/lib/libjemalloc.so)
```

Comment:
Just so I understand this correctly: Building jemalloc as a static library and linking it into hyrise statically would not work, because the weak symbol replacement stuff only works with dynamic libs? (@mrks?).

If that's not the case would it be trivially possible to build jemalloc as a static lib, for consistency with the other third party libs?

Reply:
Not even sure if it generally does not work, but we might lose compatibility with other tools that expect malloc to be weak. Think about profilers, valgrind, ...